### PR TITLE
validators for gulp, cocoapods, xcodebuild, carthage, xcpretty, libmobiledevice, deviceconsole, check-engine

### PR DESCRIPTION
**Pull request by olexme**

### CODE REVIEW

**Review by mohlsen — CHANGES_REQUESTED** [review]
looks like this is checking for the local vesion of gulp (3.9.1) which is just installed via npm locally via package.json (not installed globally).  The gulp validator should really just be checking the presence of the gulp CLI.

File: lib/validatorRules.js [hunk 1]
```diff
@@ -85,5 +85,45 @@ module.exports = {
         versionCheck: 'ver',
         versionValidate:
             (result, version) => result.includes(version)
+    },
+    gulp: {
```
Comment (mohlsen) [inline lib/validatorRules.js]: maybe rename this to gulp-cli or something to make it clear

File: lib/validatorRules.js [hunk 2]
```diff
@@ -85,5 +85,45 @@ module.exports = {
...
+        versionCheck: 'gulp -version',
+        versionValidate:
+            (result, version) => result.includes(version)
```
Comment (mohlsen) [inline lib/validatorRules.js:92]: and then just look to make sure the CLI matches.
```
[08:42:55] CLI version 1.2.1
[08:42:55] Local version 3.9.1
```

Comment (mohlsen) [reply lib/validatorRules.js:92]: might need to use a regex to look for CLI like the git validator

File: lib/validatorRules.spec.js [hunk 3]
```diff
@@ -74,4 +74,125 @@ test('validatorRules', (t) => {
         });
     });
 
+    t.test('gulp version', (t) => {
```
Comment (mohlsen) [inline lib/validatorRules.spec.js]: add a test case that has both the CLI and local version

**Comment by mohlsen** [comment]
Woot!  thanks for the PR @olexme!!  Just a couple of updates, mainly for gulp since we should check for the CLI instead of local version.

**Comment by mohlsen** [comment]
Closes #46, #38, and #36.

**Comment by olexme** [comment]
@mohlsen switched to global gulp-cli validator instead of local gulp

**Review by mohlsen — APPROVED** [review]

**Pull request merged**